lib/health_monitor/providers/database.rb

```diff
@@ -7,10 +7,30 @@ module Providers
     class DatabaseException < StandardError; end
 
     class Database < Base
+      class Configuration < Base::Configuration
+        DEFAULT_CONFIG_NAME = nil
+
+        attr_reader :config_name
+
+        def initialize(provider)
+          super
+
+          @config_name = DEFAULT_CONFIG_NAME
+        end
+
+        def config_name=(value)
+          @config_name = value.presence&.to_s
+        end
+      end
+
       def check!
+        checked = false
```

Comment:
Why do you need this?

Reply:
I wanted to raise an error, when there is no connection checked with given name, otherwise it may give you false confidence, that database is up, but actually the config name didn't match with any connection.